File: src/Vograph.Desktop/Legal/LegalDocuments.cs
namespace Vograph.Desktop.Legal;

public sealed record LegalText(string Id, string Title, string Body);

/// <summary>The same files the web client renders. Settings and account entry show these strings.</summary>
public static class LegalDocuments
{
    public const string AgreementTitle = "Пользовательское соглашение";
    public const string PolicyTitle = "Политика обработки персональных данных";

    public static LegalText Agreement { get; } = Load("agreement", AgreementTitle, "legal.user-agreement.txt");
    public static LegalText Policy { get; } = Load("policy", PolicyTitle, "legal.privacy-policy.txt");

    public static LegalText Open(string id) => id == "agreement" ? Agreement : Policy;

    private static LegalText Load(string id, string title, string name)
    {
        using var stream = typeof(LegalDocuments).Assembly.GetManifestResourceStream(name)
            ?? throw new InvalidOperationException(name);
        using var reader = new StreamReader(stream);
        var body = reader.ReadToEnd().TrimStart('\uFEFF').Trim();
        return new LegalText(id, title, body);
    }
}

File: android/app/src/main/java/ru/bgtu_voenmeh/zapara/ui/legal/LegalDocuments.kt
package ru.bgtu_voenmeh.zapara.ui.legal

import java.io.InputStream

data class LegalText(val id: String, val title: String, val body: String)

/** Opens the shared legal files. Settings passes the asset opener; tests pass the same files. */
object LegalDocuments {
    const val AGREEMENT_TITLE = "Пользовательское соглашение"
    const val POLICY_TITLE = "Политика обработки персональных данных"

    fun open(read: (String) -> InputStream, id: String): LegalText {
        val title = if (id == "agreement") AGREEMENT_TITLE else POLICY_TITLE
        val name = if (id == "agreement") "user-agreement.txt" else "privacy-policy.txt"
        val body = read(name).bufferedReader(Charsets.UTF_8).use { it.readText() }.trimStart('\uFEFF').trim()
        return LegalText(id, title, body)
    }
}
